go/mysql/flavor_filepos.go

```diff
@@ -198,15 +198,37 @@ func (flv *filePosFlavor) status(c *Conn) (SlaveStatus, error) {
 	}
 
 	status := parseSlaveStatus(resultMap)
-	pos, err := strconv.Atoi(resultMap["Exec_Master_Log_Pos"])
-	if err != nil {
+
+	execMasterLogPosStr := resultMap["Exec_Master_Log_Pos"]
+	filePos, err := strconv.Atoi(execMasterLogPosStr)
+	if err != nil && execMasterLogPosStr != "" {
 		return SlaveStatus{}, fmt.Errorf("invalid FilePos GTID (%v): expecting pos to be an integer", resultMap["Exec_Master_Log_Pos"])
 	}
+	file := resultMap["Relay_Master_Log_File"]
+	if file != "" && execMasterLogPosStr != "" {
+		status.Position.GTIDSet = filePosGTID{
+			file: file,
+			pos:  filePos,
+		}
+	}
 
-	status.Position.GTIDSet = filePosGTID{
-		file: resultMap["Relay_Master_Log_File"],
-		pos:  pos,
+	readMasterLogPosStr := resultMap["Read_Master_Log_Pos"]
+	relayPos, err := strconv.Atoi(readMasterLogPosStr)
+	if err != nil && readMasterLogPosStr != "" {
+		return SlaveStatus{}, fmt.Errorf("invalid FilePos GTID (%v): expecting pos to be an integer", resultMap["Exec_Master_Log_Pos"])
```

Comment:
To prevent these kind of mistakes in general I decided it's better to just use the already allocated variable holding the value at that key, and fixed up all implementations to use the pre-allocated var.